orderer/common/multichannel/registrar.go

```diff
@@ -1044,3 +1066,18 @@ func (r *Registrar) removeSystemChannel() error {
 
 	return nil
 }
+
+func (r *Registrar) removeLedgerAsync(channelID string) {
+	r.pendingRemoval[channelID] = true
+
+	go func() {
+		if err := r.ledgerFactory.Remove(channelID); err != nil {
+			r.pendingRemoval[channelID] = false
+			logger.Errorf("Ledger factory failed to remove empty ledger '%s', error: %s", channelID, err)
+			return
+		}
+		r.lock.Lock()
+		r.pendingRemoval[channelID] = true
```

Comment:
needs to be 
`r.pendingRemoval[channelID] = false`

or better `delete(r.pendingRemoval, channelID)` to avoid keeping all those removed channels in the map

Reply:
I used false in consideration adding some ongoing retry logic but I'll delete them for now.

Reply:
actually I meant to delete this but copied the wrong line.